pravega/src/main/scala/akka/stream/alpakka/pravega/PravegaSettings.scala

```diff
@@ -0,0 +1,292 @@
+import io.pravega.client.stream.ReaderConfig.ReaderConfigBuilder
+
+abstract class WithClientConfig(config: Config,
+                                clientConfig: Option[ClientConfig] = None,
+                                clientConfigCustomization: Option[ClientConfigBuilder => ClientConfigBuilder] = None) {
+
+  def handleClientConfig() = {
+    val rawClientConfig =
+      clientConfig.getOrElse(ConfigHelper.buildClientConfigFromTypeSafeConfig(config))
+
+    clientConfigCustomization match {
+      case Some(cust) => cust(rawClientConfig.toBuilder).build()
+      case _ => rawClientConfig
+    }
+
+  }
+
+}
+
+class ReaderSettingsBuilder(config: Config,
+                            clientConfig: Option[ClientConfig] = None,
+                            clientConfigCustomization: Option[ClientConfigBuilder => ClientConfigBuilder] = None,
+                            readerConfigBuilder: ReaderConfigBuilder,
+                            readerConfigCustomizer: Option[ReaderConfigBuilder => ReaderConfigBuilder] = None,
+                            groupName: Option[String],
+                            timeout: Long,
+                            readerId: Option[String])
+    extends WithClientConfig(config, clientConfig, clientConfigCustomization) {
+
+  def withClientConfig(clientConfig: ClientConfig): ReaderSettingsBuilder = copy(clientConfig = Some(clientConfig))
+
+  def clientConfigBuilder(clientConfigCustomization: ClientConfigBuilder => ClientConfigBuilder) =
+    copy(clientConfigCustomization = Some(clientConfigCustomization))
+
+  def readerConfigBuilder(f: ReaderConfigBuilder => ReaderConfigBuilder) = copy(readerConfigCustomizer = Some(f))
+
+  def withGroupName(name: String): ReaderSettingsBuilder = copy(groupName = Some(name))
+  def withReaderId(id: String): ReaderSettingsBuilder = copy(readerId = Some(id))
+  def withTimeout(t: Long): ReaderSettingsBuilder = copy(timeout = t)
+
+  def withSerializer[A](serializer: Serializer[A]): ReaderSettings[A] = {
+
+    readerConfigCustomizer.foreach(_(readerConfigBuilder))
+
+    new ReaderSettings[A](handleClientConfig(),
+                          readerConfigBuilder.build(),
+                          groupName.getOrElse(throw new IllegalStateException("group-name is mandatory")),
+                          timeout,
+                          serializer,
+                          readerId)
+  }
+
+  private def copy(clientConfig: Option[ClientConfig] = clientConfig,
+                   clientConfigCustomization: Option[ClientConfigBuilder => ClientConfigBuilder] =
+                     clientConfigCustomization,
+                   readerConfigBuilder: ReaderConfigBuilder = readerConfigBuilder,
+                   readerConfigCustomizer: Option[ReaderConfigBuilder => ReaderConfigBuilder] = readerConfigCustomizer,
+                   groupName: Option[String] = groupName,
+                   timeout: Long = timeout,
```

Comment:
This can be a Java Duration.  Then users can define durations in their config as `1 second`, or `500 ms`.  Lightbend config will automatically parse these fields.  It can be converted to a `long` for Pravega.